File: src/main/java/io/github/bloodnighttw/WhateverBot/WhateverBot.java
package io.github.bloodnighttw.WhateverBot;

import ch.qos.logback.classic.Level;
import ch.qos.logback.classic.Logger;
import io.github.bloodnighttw.WhateverBot.VoiceChannelAutoCraft.VCAutoCraftLoader;
import io.github.bloodnighttw.WhateverBot.utils.command.CommandRegister;
import net.dv8tion.jda.api.JDA;
import net.dv8tion.jda.api.JDABuilder;
import net.dv8tion.jda.api.events.ReadyEvent;
import net.dv8tion.jda.api.hooks.EventListener;
import org.slf4j.LoggerFactory;

import javax.security.auth.login.LoginException;

public class WhateverBot {

	private static CommandRegister commandRegister;

	public static void main(String[] args) throws LoginException {

		for (String it : args) {
			switch (it) {
				case "-d":
				case "-debug":
					Logger root = (Logger) LoggerFactory.getLogger(org.slf4j.Logger.ROOT_LOGGER_NAME);
					root.setLevel(Level.DEBUG);
					System.out.println("Enable Debug Mode");
					break;
			}
		}

		JDA bot = JDABuilder.createDefault(System.getenv("TOKEN")).build();
		commandRegister = new CommandRegister(bot);

		load(bot);

		bot.addEventListener((EventListener) event -> {
			if (event instanceof ReadyEvent) {
				commandRegister.addToAllServer();
			}
		});


	}

	static void load(JDA bot) {
		if (System.getenv("CHANNEL_ID") != null)
			bot.addEventListener(new io.github.bloodnighttw.WhateverBot.CodeAutoResend.MessageHandler());
		if (System.getenv("VC_ID") != null)
			VCAutoCraftLoader.load(bot, commandRegister);

	}
}

File: src/main/java/io/github/bloodnighttw/whateverBot/WhateverBot.kt
@file:JvmName(name = "WhateverBot")

package io.github.bloodnighttw.whateverBot

import ch.qos.logback.classic.Level
import io.github.bloodnighttw.whateverBot.codeWrapper.MessageHandler
import io.github.bloodnighttw.whateverBot.utils.command.CommandRegister
import net.dv8tion.jda.api.JDA
import net.dv8tion.jda.api.JDABuilder
import org.slf4j.LoggerFactory
import org.yaml.snakeyaml.Yaml
import java.io.File
import java.io.InputStream
import kotlin.system.exitProcess


fun load(bot: JDA) {
	val commandRegister = CommandRegister(bot)
	bot.addEventListener(MessageHandler())
	commandRegister.addToAllServer()
}

fun main(args: Array<String>) {

	var configFileLocation = "./config.yaml"

	for (i in args.indices) {
		when (args[i]) {
			"-d", "--debug" -> run {
				val root = LoggerFactory.getLogger(org.slf4j.Logger.ROOT_LOGGER_NAME) as ch.qos.logback.classic.Logger
				root.level = Level.DEBUG
			}

			"-c", "--config" -> {
				configFileLocation = args[i+1];
			}

		}
	}

	System.getenv("TOKEN")?.let {
		val bot = JDABuilder.createDefault(it)
				.build().awaitReady()
		load(bot)
	} ?: run {
		val s = "Please Enter Your Discord Bot Token in config.yaml"
		println()
		println(s)
	}



}